src/circuit/index.js

```diff
@@ -1,197 +1,87 @@
-    const addrs = ma.toString().split('/p2p-circuit')
-    const relayAddr = multiaddr(addrs[0])
-    const destinationAddr = multiaddr(addrs[addrs.length - 1])
-    const relayPeer = PeerId.createFromCID(relayAddr.getPeerId())
-    const destinationPeer = PeerId.createFromCID(destinationAddr.getPeerId())
-
-    let disconnectOnFailure = false
-    let relayConnection = this._connectionManager.get(relayPeer)
-    if (!relayConnection) {
-      relayConnection = await this._dialer.connectToPeer(relayAddr, options)
-      disconnectOnFailure = true
-    }
-
-    try {
-      const virtualConnection = await hop({
-        connection: relayConnection,
-        circuit: this,
-        request: {
-          type: CircuitPB.Type.HOP,
-          srcPeer: {
-            id: this.peerId.toBytes(),
-            addrs: this._libp2p.multiaddrs.map(addr => addr.bytes)
-          },
-          dstPeer: {
-            id: destinationPeer.toBytes(),
-            addrs: [multiaddr(destinationAddr).bytes]
-          }
-        }
-      })
-
-      const localAddr = relayAddr.encapsulate(`/p2p-circuit/p2p/${this.peerId.toB58String()}`)
-      const maConn = toConnection({
-        stream: virtualConnection,
-        remoteAddr: ma,
-        localAddr
-      })
-      log('new outbound connection %s', maConn.remoteAddr)
-
-      return this._upgrader.upgradeOutbound(maConn)
-    } catch (err) {
-      log.error('Circuit relay dial failed', err)
-      disconnectOnFailure && await relayConnection.close()
-      throw err
+  start () {
+    // Advertise service if HOP enabled
+    const canHop = this._options.hop.enabled
+
+    if (canHop && this._options.advertise.enabled) {
+      this._timeout = setTimeout(() => {
+        this._advertiseService()
+      }, this._options.advertise.bootDelay)
     }
   }
 
   /**
-   * Create a listener
-   *
-   * @param {any} options
-   * @param {Function} handler
-   * @return {listener}
+   * Stop Relay service.
+   * @returns {void}
    */
-  createListener (options, handler) {
-    if (typeof options === 'function') {
-      handler = options
-      options = {}
-    }
-
-    // Called on successful HOP and STOP requests
-    this.handler = handler
-
-    return createListener(this._libp2p, options)
+  stop () {
+    clearTimeout(this._timeout)
   }
 
   /**
-   * Filter check for all Multiaddrs that this transport can dial on
-   *
-   * @param {Array<Multiaddr>} multiaddrs
-   * @returns {Array<Multiaddr>}
+   * Advertise hop relay service in the network.
+   * @returns {Promise<void>}
    */
-  filter (multiaddrs) {
-    multiaddrs = Array.isArray(multiaddrs) ? multiaddrs : [multiaddrs]
+  async _advertiseService () {
+    try {
+      const cid = await namespaceToCid(RELAY_RENDEZVOUS_NS)
+      await this._libp2p.contentRouting.provide(cid)
+    } catch (err) {
+      if (err.code === 'NO_ROUTERS_AVAILABLE') {
+        log('there are no routers configured to advertise hop relay service')
```

Comment:
I think this should be an error since you can disable advertising. If I'm running a "private" relay, I should turn off advertising and shouldn't need to be running a DHT. I would also clarify the error message and log the error as well.
```suggestion
        log.error('a content router, such as a DHT, must be provided in order to advertise the relay service', err)
```